app.js

```diff
@@ -28,9 +31,65 @@ app.use(express.urlencoded({ extended: true }));
 app.use(session(sess));
 
 app.use(express.json());
-app.use("/api/v1/query-info", v1QueryInfo);
+app.get("/api/v1/engines", async (req, res) => {
+    const path = "engines"
+    const folders = fs.readdirSync(path).filter(f => fs.lstatSync(`${path}/${f}`).isDirectory());
+
+    res.send({
+        success: true, 
+        engines: folders 
+    });
+    return;
+})
+
+app.get("/api/v1/engines/:engine/parameters", async (req, res) => {
+  const engine = await import(`./engines/${req.params.engine}/engine.js`)
+  const baseParameters = {
+    "prompt": "type=string, required=true, description of desired model or changes to model",
+    "format": "type=string, default='sd-json', options are <sd-json|xmile>",
```

Comment:
I think for now we stick with JSON only.  Lets wait for someone to really need the XMILE before we write a generic XMILE parser in JS.